### PR TITLE
WalletConnect v2.0 integration, remove WalletConnect v1.0 from UI

### PR DESCRIPTION
**Pull request by smitch88**
#1632 

**Description:**
* Add support for WalletConnect v2.0 which replaces WalletConnect v1.0 in the UI.
* Change out all `web3React/core` usages in favor of RainbowKit + Wagmi
* Adds Wagmi and RainbowKit providers
* Needed to upgrade NextJS 10 -> 11, to support build, needed to move old babel config (its kept because of `fbt` library), upgraded to webpack 5 build

**Changes:**

The change spans across both dapps, OUSD, and OETH.

* Add in Rainbow kit wallet integration + associated wallet connectors to match previous support
* Kept `Wagmi` on version v0.12 to avoid upgrading to `viem` from `ethers`
* Updates all references to web3react to the equivalent `wagmi` hooks

~* There are package management changes and required dependency upgrades for NextJS `10.3.0 -> 11.1.0` and `webpack 4` -> `webpack 5` to support what we need to include `walletconnect/ethereum-provider@2.8.0` and work within the NextJS build. This required moving to .babelrc, and some other changes. I had to also alias the v8 version of `@web3-react/core` -> `web3-react-v8`, so we could be backward compatible with the existing connectors from 2+ years ago.~

~* Removed "My Ether Wallet" support due to compatibility issues with the upgrades of other packages. Confirmed with product via @micahalcorn that we can revisit this at a later date.~

**Required:**
* Add into `.env`, `NEXT_PUBLIC_WALLETCONNECT_V2_PROJECT_ID`, with walletconnect from `clout.walletconnect.com` project id

**Screenshots**

## OETH
Screenshot 2023-07-03 131921

### CODE REVIEW

File: contracts/test/flipper/flipper.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ const {
   loadFixture,
   isFork,
 } = require("../helpers");
-const { parseUnits } = require("ethers/lib/utils");
```
Comment (rafaelugolini) [inline contracts/test/flipper/flipper.js:11]: line 106 is broken with this change, the linter got it

File: dapp-oeth/src/components/TransactionListener.js [hunk 2]
```diff
@@ -272,7 +271,9 @@ const TransactionListener = ({ rpcProvider }) => {
         // TODO handle a 404 here. We need to retry. A refresh is required to
         // get this to retry at the moment.
         try {
-          safeData = await connector.sdk.txs.getBySafeTxHash(t.hash)
+          // TODO
+          console.log(connector)
```
Comment (dcodes05) [inline dapp-oeth/src/components/TransactionListener.js:275]: this console.log should be removed, right? it is not needed for users, right?

File: dapp/src/components/TransactionListener.js [hunk 3]
```diff
@@ -272,7 +271,9 @@ const TransactionListener = ({ rpcProvider }) => {
         // TODO handle a 404 here. We need to retry. A refresh is required to
         // get this to retry at the moment.
         try {
-          safeData = await connector.sdk.txs.getBySafeTxHash(t.hash)
+          // TODO
+          console.log(connector)
```
Comment (dcodes05) [inline dapp/src/components/TransactionListener.js:275]: same thing, do you need console.log in prod?

**Review by dcodes05 — COMMENTED** [review]
there are a couple of console.logs that should be removed

File: dapp/src/components/TransactionListener.js (hunk 3 above)
Comment (smitch88) [reply dapp/src/components/TransactionListener.js:275]: Correct, this actually isn't done yet. I need to adjust this equivalent from web3 react. Left a TODO, revisiting it, good call out though.

File: dapp-oeth/src/components/TransactionListener.js (hunk 2 above)
Comment (smitch88) [reply dapp-oeth/src/components/TransactionListener.js:275]: Same comment as the other. Need to find the equivalent of this call from web3-react -> wagmi. The library switch had a gap between them

File: dapp/src/components/TransactionListener.js (hunk 3 above)
Comment (rafaelugolini) [reply dapp/src/components/TransactionListener.js:275]: @smitch88 does it mean that it's not ready to be merged?

Comment (smitch88) [reply dapp/src/components/TransactionListener.js:275]: This is just an issue with transaction display for gnosis specifically. I think we could merge and unblock most of things while fixing this as a follow up.

**Review by micahalcorn — APPROVED** [review]

**Comment by micahalcorn** [comment]
We've tested enough on the OETH side to feel good about this and fix forward. OUSD's dapp will be tested in staging.

**Pull request merged**